app/controllers/lettings_logs_controller.rb

```diff
@@ -111,6 +110,37 @@ def update_logs
     end
   end
 
+  def delete_logs_list
+    all_logs = current_user.lettings_logs.visible
+    @search_term = search_term
+    @delete_logs_form = Forms::DeleteLogsForm.new(logs: filtered_logs(all_logs, @search_term, @session_filters))
+    if params[:no_logs_selected] == "true"
+      @delete_logs_form.errors.add(:logs_ids, "Select at least one log to delete or press cancel to return")
+    end
+    render "logs/delete_logs"
+  end
+
+  def delete_logs_confirmation
+    log_ids = params.dig(:forms_delete_logs_form, :logs_to_delete)
+    log_type = params.dig(:forms_delete_logs_form, :log_type).to_sym
+    @delete_logs_form = Forms::DeleteLogsForm.new(log_ids:, log_type:)
+    @search_term = params.dig(:forms_delete_logs_form, :search_term)
+    if @delete_logs_form.no_logs_selected?
+      redirect_to delete_logs_lettings_logs_path(search: @search_term, no_logs_selected: true)
+    end
+  end
+
+  def delete_logs
+    ids_to_delete = params.require("ids").split.map(&:to_i)
+    ids_to_delete.each do |id|
+      log = LettingsLog.visible.find(id)
+      authorize log, :destroy?
+      log.discard!
+    end
```

Comment:
I'm curious why we need to split IDs here?

```suggestion
  def delete_logs
    logs = LettingsLog.visible.find(params[:ids])
    logs.each { |log| authorize(log, :destroy?) }
    logs.discard_all!
  end
```